### PR TITLE
Mf/bidder input edge case testing

### PR DESCRIPTION
**Pull request by mattfoley8**
This PR tests for the following Bidder-Input edge cases:

Consider a new order submitted by m1 buying $DESO and selling m1 DAO coins that will match an existing order submitted by m0 buying m1 DAO coins and selling $DESO. Since m0 is selling $DESO, m1 needs to specify BidderInputs including m0's UTXOs to cover the $DESO transferred.
* m1 submits no BidderInputs. Fails.
* m1 submits their own UTXOs as BidderInputs. Fails.
* m1 submits m2's UTXOs as BidderInputs. Fails.
* m1 submits spent m0 UTXOs as BidderInputs. Fails.
* m1 submits m0's UTXOs and their own UTXOs as BidderInputs. Fails.
* m1 submits m0's UTXOs and m2's UTXOs as BidderInputs. Fails.
* m1 submits m0's correct UTXOs. Passes.

Consider a new order submitted by m1 buying m0 DAO coins and selling m1 DAO coins that will match an existing order submitted by m0 buying m1 DAO coins and selling m0 DAO coins. Here, no BidderInputs should be specified as this is a DAO<-->DAO coin trade.
* m1 submits m0's UTXOs as BidderInputs. Fails.
* m1 submits no UTXOs as BidderInputs. Passes.

### CODE REVIEW

File: lib/block_view_dao_coin_limit_order_test.go [hunk 1]
```diff
@@ -1732,6 +1731,260 @@ func TestDAOCoinLimitOrder(t *testing.T) {
...
+				TransactorPublicKey: NewPublicKey(m1PkBytes),
+				Inputs:              append([]*DeSoInput{}, (*DeSoInput)(utxoEntriesM2[0].UtxoKey)),
+			})
+
+		txnMeta.BidderInputs = bidderInputs
+
+		_, _, _, _, err = _connectDAOCoinLimitOrderTxn(
+			testMeta, m1Pub, m1Priv, currentTxn, totalInputMake)
+		require.Error(err)
+		require.Contains(err.Error(), RuleErrorDAOCoinLimitOrderFeeNanosBelowMinTxFee)
```
Comment (lazynina) [inline lib/block_view_dao_coin_limit_order_test.go:1895]: If we doubled or tripled the FeeRateNanosPerKB would we still hit this error? Or would it go through and give M2 their money back?

File: lib/block_view_dao_coin_limit_order_test.go [hunk 2]
```diff
@@ -1732,6 +1731,260 @@ func TestDAOCoinLimitOrder(t *testing.T) {
...
+			[]*DeSoInputsByTransactor{},
+			&DeSoInputsByTransactor{
+				TransactorPublicKey: NewPublicKey(m1PkBytes),
+				Inputs:              append([]*DeSoInput{}, (*DeSoInput)(utxoEntriesM0[0].UtxoKey)),
+			})
+
+		_, _, _, _, err = _connectDAOCoinLimitOrderTxn(
+			testMeta, m1Pub, m1Priv, currentTxn, totalInputMake)
+		require.Error(err)
+		require.Contains(err.Error(), RuleErrorDAOCoinLimitOrderFeeNanosBelowMinTxFee)
```
Comment (lazynina) [inline lib/block_view_dao_coin_limit_order_test.go:1972]: Same as above, if we increase the fee rate, will this go through and give M0 their money back?

File: lib/block_view_dao_coin_limit_order_test.go (hunk 1 above)
Comment (mattfoley8) [reply lib/block_view_dao_coin_limit_order_test.go:1895]: I was using m1's PK. Now passes and UTXOs are refunded. I left in an additional check for the above where I use m1's PK but m2's UTXO to continue to test for the above error.

File: lib/block_view_dao_coin_limit_order_test.go [hunk 3]
```diff
@@ -1732,6 +1731,356 @@ func TestDAOCoinLimitOrder(t *testing.T) {
...
+
+		// m1 includes m0's BidderInputs in addition to
+		// m2's and tries to connect. Should error.
+		bidderInputs = append([]*DeSoInputsByTransactor{}, originalBidderInput)
+
+		bidderInputs = append(
+			bidderInputs,
+			&DeSoInputsByTransactor{
+				TransactorPublicKey: NewPublicKey(m1PkBytes),
```
Comment (mattfoley8) [inline lib/block_view_dao_coin_limit_order_test.go:1886]:
```suggestion
				TransactorPublicKey: NewPublicKey(m2PkBytes),
```

Comment (mattfoley8) [reply lib/block_view_dao_coin_limit_order_test.go:1886]: We can optionally make this change. Either way, should result in the same error.

Comment (mattfoley8) [reply lib/block_view_dao_coin_limit_order_test.go]: Made the change below.

**Review by lazynina — APPROVED** [review]
